calyx-opt/src/passes/static_inliner.rs

```diff
@@ -0,0 +1,388 @@
+                    StaticInliner::update_assignment_timing(
+                        &mut g_assigns,
+                        cur_offset,
+                        stmt_latency,
+                    );
+                    // add g_assigns to seq_group_assigns
+                    seq_group_assigns.extend(g_assigns.into_iter());
+                    // updates cur_offset so that next stmt gets its static timign
+                    // offset appropriately
+                    cur_offset += stmt_latency;
+                }
+                assert!(
+                    *latency == cur_offset,
+                    "static group latency doesn't match static seq latency"
+                );
+                seq_group.borrow_mut().assignments = seq_group_assigns;
+                seq_group.borrow_mut().attributes = attributes.clone();
+                seq_group
+            }
+            ir::StaticControl::Par(ir::StaticPar {
+                stmts,
+                latency,
+                attributes,
+            }) => {
+                let par_group =
+                    builder.add_static_group("static_par", *latency);
+                let mut par_group_assigns: Vec<
+                    ir::Assignment<ir::StaticTiming>,
+                > = vec![];
+                for stmt in stmts {
+                    let stmt_latency = stmt.get_latency();
+                    // recursively turn each stmt in the par block into a group g
+                    let g = StaticInliner::inline_static_control(stmt, builder);
+                    assert!(
+                        g.borrow().get_latency() == stmt_latency,
+                        "static group latency doesn't match static stmt latency"
+                    );
+                    // get the assignments from g
+                    // see note on the StaticControl::Seq(..) case abt why we need to clone
+                    let mut g_assigns: Vec<ir::Assignment<ir::StaticTiming>> =
+                        g.borrow_mut().assignments.clone();
+                    // offset = 0 (all start at beginning of par),
+                    // but still should add %[0:stmt_latency] to beginning of group
+                    StaticInliner::update_assignment_timing(
+                        &mut g_assigns,
+                        0,
+                        stmt_latency,
+                    );
+                    // add g_assigns to par_group_assigns
+                    par_group_assigns.extend(g_assigns.into_iter());
+                }
+                par_group.borrow_mut().assignments = par_group_assigns;
+                par_group.borrow_mut().attributes = attributes.clone();
+                par_group
+            }
+            ir::StaticControl::If(ir::StaticIf {
+                port,
+                tbranch,
+                fbranch,
+                latency,
+                attributes,
+            }) => {
+                let if_group = builder.add_static_group("static_if", *latency);
```

Comment:
Is the `latency` of a `StaticIf` the max of the latencies of the two branches?

Reply:
yes.